root/ntuple/ntuple-makeproject/CMakeLists.txt

```
@@ -0,0 +1,54 @@
+if(NOT ROOT_root7_FOUND)
+  return()
+endif()
+
+# Generate common dictionary for custom class
+ROOT_GENERATE_DICTIONARY(eventDict ${CMAKE_CURRENT_SOURCE_DIR}/ntuple_makeproject_header.h LINKDEF ntuple_makeproject_header_LinkDef.h)
+
+# MakeProject with TTree data
+# Write the TTree with the custom class branch
+ROOTTEST_GENERATE_EXECUTABLE(write_ttree write_ttree.cxx eventDict.cxx LIBRARIES Core RIO Tree FIXTURES_SETUP write_ttree_executable)
+ROOTTEST_ADD_TEST(write_ttree
+                  EXEC ./write_ttree
+                  FIXTURES_REQUIRED write_ttree_executable
+                  FIXTURES_SETUP written_ttree)
+
+# Call MakeProject on the output file
+ROOTTEST_GENERATE_EXECUTABLE(makeproject_ttree makeproject_ttree.cxx LIBRARIES Core RIO Tree FIXTURES_REQUIRED written_ttree FIXTURES_SETUP makeproject_ttree)
+ROOTTEST_ADD_TEST(makeproject_ttree
+                  EXEC ./makeproject_ttree
+                  FIXTURES_REQUIRED makeproject_ttree
+                  FIXTURES_SETUP makeproject_ttree_called)
```

```suggestion
ROOTTEST_GENERATE_EXECUTABLE(makeproject_ttree makeproject_ttree.cxx LIBRARIES Core RIO Tree FIXTURES_SETUP makeproject_ttree)
ROOTTEST_ADD_TEST(makeproject_ttree
                  EXEC ./makeproject_ttree
                  FIXTURES_REQUIRED "makeproject_ttree;written_ttree"
                  FIXTURES_SETUP makeproject_ttree_called)
```
A priori the linking of `makeproject_ttree` does not require the root file to have been written yet.

---

If this is right, this would apply to the other executables.

---

Yes you're right thanks, I have updated the commit